src/gui/qgsmapoverviewcanvas.cpp

```diff
@@ -249,11 +250,21 @@ void QgsMapOverviewCanvas::setLayers( const QList<QgsMapLayer *> &layers )
 
 void QgsMapOverviewCanvas::updateFullExtent()
 {
+  QgsReferencedRectangle extent = QgsProject::instance()->viewSettings()->fullExtent();
```

Comment:
The way I read it, if a user hasn't specified a project extent, this will return the extent of the project layers' combined extents? If so, this will create an UX regression as the overview panel does _not_ use the map canvas extent when layer(s) are shown in the overview but rather the extent of only those overview-drawn layers.

So the logic here should be:
_user-set_ project extent, if not overview-drawn layer combined extent, if not map canvas extent.

Reply:
How's it look now?

Reply:
Code logic looks good, thanks.